### DISCUSSION TITLE
Usage of rotate_log api command

### DISCUSSION THREAD

**Discussion by Millesimus** [processone/ejabberd#4156] [Questions] [answered]
Hey everyone,

I'm not sure whether this is an error in configuration or in understanding.

I tried to rotate my logs with the `rotate_log` api command, because I thought that this command would be a manual trigger to rotate the logs (while the log config options would be for automatic log rotation). However, and although the command is executed successfully, nothing happens and the logs are not rotating.

Have I misunderstood how the command is supposed to work? Or may I have an error in my config?

**Comment by badlop** [comment, marked as answer]
You are right, the rotate_log and reopen_log commands documentation make seem that they immediately perform those operations, but looking at the source code and its history, they are no longer relevant for those operations!

Commit 07baf2d9732c442ab21cc618266da8bca4864e0a for ejabberd 16.01 introduced the rotate_log API command, and it worked as you expected.

Commit e4a8afb15d6c995b294a60ed150b86775e2f6805 for ejabberd 20.01 replaced the old Lager library with Logger, a new library included in Erlang/OTP 21. Since then, rotate_log and reopen_log commands no longer rotate or reopen, they just run ejabberd hooks:
- rotate_log: I couldn't find any usage in ejabberd or ejabberd-contrib... maybe it's used by other modules out there
- reopen_log: used by mod_http_fileserver, mod_logsession, mod_message_log, mod_s2s_log and mod_spam_filter

It should get mentioned in the commands documentation that they lost their original purpose

https://docs.ejabberd.im/developer/ejabberd-api/admin-api/#rotate-log

**Reply by Millesimus** [reply]
Thank you for effort investigating the functionality! It's good to know that this is the expected behaviour (and sorry for the late reply).
